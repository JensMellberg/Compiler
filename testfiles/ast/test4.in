int a () {
  x = 2 == 1 + 1;
  x = (1 == 1) * 2;

  x = (1 == 2) == 5;   //GOOD CODE

  x = (5 > 2) + (2 < 5);
}
