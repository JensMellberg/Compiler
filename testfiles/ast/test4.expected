Program
  List
    FuncDecl
      IdDecl ID="a"
      List
      Block
        List
          Assign
            IdUse ID="x"
            Eq
              Numeral NUMERAL="2"
              Add
                Numeral NUMERAL="1"
                Numeral NUMERAL="1"
          Assign
            IdUse ID="x"
            Mul
              Eq
                Numeral NUMERAL="1"
                Numeral NUMERAL="1"
              Numeral NUMERAL="2"
          Assign
            IdUse ID="x"
            Eq
              Eq
                Numeral NUMERAL="1"
                Numeral NUMERAL="2"
              Numeral NUMERAL="5"
          Assign
            IdUse ID="x"
            Add
              Gtr
                Numeral NUMERAL="5"
                Numeral NUMERAL="2"
              Lss
                Numeral NUMERAL="2"
                Numeral NUMERAL="5"
